articles/quasar-google-auth-with-firebase/index.md

```diff
@@ -0,0 +1,640 @@
+        </q-tab-panel>
+
+        <q-tab-panel name="register">
+          <AuthComponent :tab="tab"/>
+        </q-tab-panel>
+      </q-tab-panels>
+    </q-card>
+  </q-page>
+</template>
+```
+
+The `AuthComponent` above is imported to the `pages/Auth.vue` with props for register and login tabs. In the `<script>` tag paste the following code:
+
+```JavaScript
+import AuthComponent from "components/AuthComponent";
+export default {
+  components: { AuthComponent },
+  data () {
+    return {
+      tab: 'login'
+    }
+  }
+}
+```
+
+### Add Firebase to our application
+
+To add firebase to our app, ensure the following is done correctly:
+
+1. Visit`[firebase console](https://console.firebase.google.com/u/0/)
+2. Click `add project` to create a firebase project to connect to our app. Name it `quasar-google-auth` and click `Continue`
+3. Disable Google Analytics for this project and click `Continue` to create our project.
+4. After the project is ready click `Continue`. It will take you to the project overview. Click the `web` icon, then register an app name, name it `quasar-firebase-googleauth` and click `Register App`
+5. You'll be taken to `step 2 - Add Firebase SDK`, copy the whole script.
+
+We're going to install firebase and create a boot file to initialize firebase before our app runs.
+Go back to our project, open the terminal, run the command below:
+
+```bash
+yarn add firebase
+quasar new boot firebase
+```
+
+Navigate to `quasar.conf.js` file, search for `boot` it should be an empty array. Add `firebase.js` boot file in it as indicated below.
+
+```JavaScript
+boot: ['firebase'],
+```
+
+Navigate to the `boot` folder, open `firebase.js` file. 
+
+Paste the script you copied the previous step for the firebase SDK. `firebase.js` file structure should resemble the one below.
+
+```JavaScript
+import firebase from "firebase";
+
+const firebaseConfig = {
+    apiKey: "",
+    authDomain: "",
+    projectId: "",
+    storageBucket: "",
+    messagingSenderId: "",
+    appId: ""
+  };
+  firebase.initializeApp(firebaseConfig);
+
+export default firebase
+```
+
+### Access Control for Routes
+
+We're going to disable access to routes which requires authorization when a user needs to access them. This is quite easy. Navigate to `routes` folder, open `index.js` file.
+
+`Import firebase from "firebase"` at the top of the file just like we did in the `firebase.js` boot file. When done, add the following code just above the line `return Router`:
+
+```JavaScript
+  Router.beforeEach(async (to, from, next) => {
+    const auth = to.meta.requiresAuth
+    if (auth && !await firebase.getCurrentUser()) {
+      next('/');
+    } else {
+      next();
+    }
+  })
+```
+
+Now if the currentUser is null or undefined, we should redirect users to the auth path (`/`). But how do we get currentUser? We can’t use `firebase.auth().currentUser` because on page refresh that property has not been set yet before the requiresAuth guard is triggered. 
```

Comment:
```suggestion
Now if the currentUser is null or undefined, we should redirect users to the auth path (`/`). But how do we get currentUser? We can’t use `firebase.auth().currentUser` because on page refresh that property has not been set yet before the `requiresAuth` guard is triggered. 
```